Parse/src/main/java/com/parse/ParseQuery.java

```diff
@@ -960,33 +960,21 @@ public ParseQuery(ParseQuery<T> query) {
     return ParseUser.getCurrentUserAsync();
   }
 
-  private void checkIfRunning() {
-    checkIfRunning(false);
-  }
-
-  private void checkIfRunning(boolean grabLock) {
-    synchronized (lock) {
-      if (isRunning) {
-        throw new RuntimeException(
-            "This query has an outstanding network connection. You have to wait until it's done.");
-      } else if (grabLock) {
-        isRunning = true;
-        cts = Task.create();
-      }
-    }
-  }
-
   /**
    * Cancels the current network request (if one is running).
    */
   //TODO (grantland): Deprecate and replace with CancellationTokens
```

Comment:
Now the cancellation is handled by the SDK itself. I think this suggestion means using the `CancellationTokenSource`/`CancellationToken` to transfer the cancellation handler to Bolt. But it is a big work, bunch of related files need to be modified. Another PR to do this job is preferred.

https://github.com/BoltsFramework/Bolts-Android#cancelling-tasks